tests/zquantum/core/evolution_test.py

```diff
@@ -1,154 +1,152 @@
-        self.assertEqual(
-            compare_unitary(final_unitary, target_unitary, tol=1e-10), True
-        )
+def _cirq_exponentiate_pauli_term(term, qubits, time, trotter_order):
+    base_exponent = 2 * time / trotter_order / np.pi
+    base_gate = PAULI_STRING_TO_CIRQ_GATE[term.pauli_string()](*qubits)
+    return base_gate ** (term.coefficient * base_exponent)
 
-    def test_time_evolution_with_symbolic_parameter(self):
-        # Given
-        hamiltonian = PauliSum(
-            [
-                PauliTerm("X", 0) * PauliTerm("X", 1),
-                PauliTerm("Y", 0, 0.5) * PauliTerm("Y", 1),
-                PauliTerm("Z", 0, 0.3) * PauliTerm("Z", 1),
-            ]
-        )
-        time_symbol = sympy.Symbol("t")
-        time_value = 0.4
-        symbols_map = [(time_symbol, time_value)]
-        order = 2
-
-        circuit = cirq.Circuit()
-        q1 = cirq.LineQubit(0)
-        q2 = cirq.LineQubit(1)
-        for _ in range(0, order):
-            circuit.append(
-                cirq.XX(q1, q2)
-                ** (hamiltonian.terms[0].coefficient * 2 * time_value / order / pi)
-            )
-            circuit.append(
-                cirq.YY(q1, q2)
-                ** (hamiltonian.terms[1].coefficient * 2 * time_value / order / pi)
-            )
-            circuit.append(
-                cirq.ZZ(q1, q2)
-                ** (hamiltonian.terms[2].coefficient * 2 * time_value / order / pi)
-            )
-        target_unitary = circuit._unitary_()
 
-        # When
-        unitary_evolution_symbolic = time_evolution(
-            hamiltonian, time_symbol, trotter_order=order
-        )
-        unitary_evolution = unitary_evolution_symbolic.evaluate(symbols_map)
-        final_unitary = unitary_evolution.to_unitary()
-        # Then
-        self.assertEqual(
-            compare_unitary(final_unitary, target_unitary, tol=1e-10), True
-        )
+def _cirq_exponentiate_qubit_hamiltonian_term(term, qubits, time, trotter_order):
+    base_exponent = 2 * time / trotter_order / np.pi
+    base_gate = OPENFERMION_TERM_TO_CIRQ_GATE[list(term.terms.keys())[0]](*qubits)
+    coefficient = list(term.terms.values())[0]
+    return base_gate ** (coefficient * base_exponent)
 
-    def test_time_evolution_derivatives(self):
-        # Given
-        hamiltonian = PauliSum(
+
+def _cirq_exponentiate_hamiltonian(hamiltonian, qubits, time, trotter_order):
+    if isinstance(hamiltonian, QubitOperator):
+        return cirq.Circuit(
             [
-                PauliTerm("X", 0) * PauliTerm("X", 1),
-                PauliTerm("Y", 0, 0.5) * PauliTerm("Y", 1),
-                PauliTerm("Z", 0, 0.3) * PauliTerm("Z", 1),
+                _cirq_exponentiate_qubit_hamiltonian_term(
+                    term, qubits, time, trotter_order
+                )
+                for term in list(hamiltonian.get_operators())
             ]
+            * trotter_order
         )
+    elif isinstance(hamiltonian, PauliSum):
```

Comment:
This is not that important in a test helper like this, but in general: try to avoid functions that sometimes return something and sometimes they don't (and hence implicitly return `None`). Here, you might want to change `elif` to `else`, so that the function fails explicitly in case of type mismatch, or raise an exception in an `else` clause.